dl/src/main/scala/com/intel/analytics/bigdl/models/autoencoder/Train.scala

```diff
@@ -24,19 +24,23 @@ import com.intel.analytics.bigdl.dataset.image._
 import com.intel.analytics.bigdl.nn.{MSECriterion, Module}
 import com.intel.analytics.bigdl._
 import com.intel.analytics.bigdl.optim._
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
 import com.intel.analytics.bigdl.utils.{Engine, T}
 import org.apache.log4j.{Level, Logger}
 import org.apache.spark.SparkContext
 import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric._
 
+import scala.reflect.ClassTag
+
 object toAutoencoderBatch {
   def apply(): toAutoencoderBatch[Float] = new toAutoencoderBatch[Float]()
 }
 
-class toAutoencoderBatch[T] extends Transformer[MiniBatch[T], MiniBatch[T]] {
+class toAutoencoderBatch[T: ClassTag]
+(implicit ev: TensorNumeric[T]) extends Transformer[MiniBatch[T], MiniBatch[T]] {
   override def apply(prev: Iterator[MiniBatch[T]]): Iterator[MiniBatch[T]] = {
     prev.map(batch => {
-      MiniBatch(batch.data, batch.data)
+      MiniBatch[T](batch.data, batch.data)
```

Comment:
Inside MiniBatch class, there is a NarrowTable being used to implement a narrow method for table. Thus, the MiniBatch requires a type.